### PR TITLE
Build changes to support jdk11+

### PR DESCRIPTION
**Pull request by agilob**
@wing328 @jimschubert 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR. Should we target 6.0.x branch for the change?

**Comment by wing328** [comment]
cc @OpenAPITools/generator-core-team

**Comment by agilob** [comment]
dump @OpenAPITools/generator-core-team

File: modules/openapi-generator-gradle-plugin/build.gradle [hunk 1]
```diff
@@ -208,13 +210,17 @@ javadoc {
...
+//        }
+//    }
+//    releaseRepository{
+//        onlyIf {
+//            nexusPublishing.useStaging.any()
+//        }
+//    }
+//}
+
+//nexusPublishing.dependsOn 'check'
```
Comment (wing328) [inline modules/openapi-generator-gradle-plugin/build.gradle:226]: May I know why these are commented out?

Comment (agilob) [reply modules/openapi-generator-gradle-plugin/build.gradle:226]: Ah yes, good spot, basically `nexusPublishing` task no longer exists, I am not sure which one should replace it. Here is the list of available publishing tasks:

```
closeAndReleaseSonatypeStagingRepository - Closes and releases open staging repository in 'sonatype' Nexus instance.
closeAndReleaseStagingRepository - Closes and releases open staging repositories in the following Nexus instance: 'sonatype'
closeSonatypeStagingRepository - Closes open staging repository in 'sonatype' Nexus instance.
generateMetadataFileForMavenJavaPublication - Generates the Gradle metadata file for publication 'mavenJava'.
generateMetadataFileForOpenApiGeneratorPluginMarkerMavenPublication - Generates the Gradle metadata file for publication 'openApiGeneratorPluginMarkerMaven'.
generateMetadataFileForPluginMavenPublication - Generates the Gradle metadata file for publication 'pluginMaven'.
generatePomFileForMavenJavaPublication - Generates the Maven POM file for publication 'mavenJava'.
generatePomFileForOpenApiGeneratorPluginMarkerMavenPublication - Generates the Maven POM file for publication 'openApiGeneratorPluginMarkerMaven'.
generatePomFileForPluginMavenPublication - Generates the Maven POM file for publication 'pluginMaven'.
publish - Publishes all publications produced by this project.
publishAllPublicationsToSonatypeRepository - Publishes all Maven publications produced by this project to the sonatype repository.
publishMavenJavaPublicationToMavenLocal - Publishes Maven publication 'mavenJava' to the local Maven repository.
publishMavenJavaPublicationToSonatypeRepository - Publishes Maven publication 'mavenJava' to Maven repository 'sonatype'.
publishOpenApiGeneratorPluginMarkerMavenPublicationToMavenLocal - Publishes Maven publication 'openApiGeneratorPluginMarkerMaven' to the local Maven repository.
publishOpenApiGeneratorPluginMarkerMavenPublicationToSonatypeRepository - Publishes Maven publication 'openApiGeneratorPluginMarkerMaven' to Maven repository 'sonatype'.
publishPluginMavenPublicationToMavenLocal - Publishes Maven publication 'pluginMaven' to the local Maven repository.
publishPluginMavenPublicationToSonatypeRepository - Publishes Maven publication 'pluginMaven' to Maven repository 'sonatype'.
publishToMavenLocal - Publishes all Maven publications produced by this project to the local Maven cache.
publishToSonatype - Publishes all Maven publications produced by this project to the 'sonatype' Nexus repository.
releaseSonatypeStagingRepository - Releases closed staging repository in 'sonatype' Nexus instance.
```

Probably `publishToSonatype`, as description is most similar to `pubishToNexus`:

(old description) `publishToNexus - Publishes all Maven publications produced by this project to Nexus.`

Comment (wing328) [reply modules/openapi-generator-gradle-plugin/build.gradle:226]:
```
publishToSonatype - Publishes all Maven publications produced by this project to the 'sonatype' Nexus repository.
```

Yes, please use it as a replacement of `nexusPublishing`

Comment (agilob) [reply modules/openapi-generator-gradle-plugin/build.gradle]: That's so strange, when I run `gradle tasks` it finds `publishToSonatype` but when I run it it crashes as task not found. `publish` does the same thing, so used it

Comment (wing328) [reply modules/openapi-generator-gradle-plugin/build.gradle]: I'll try to test it over the weekend.

**Comment by agilob** [comment]
@wing328 dont want to rush you with checking this, but could we have these changes in 6.0 please?

**Comment by wing328** [comment]
@agilob thanks again for the PR 🙏 

When you've time, can you please merge the latest master into this branch?

The Gradle publishing workflow has been updated by the Gradle team in the latest master 👍

**Comment by wing328** [comment]
@agilob I've submitted https://github.com/OpenAPITools/openapi-generator/pull/15553 based on this change.

Closing this one for the time being.

Please review https://github.com/OpenAPITools/openapi-generator/pull/15553 when you've time.

**Pull request closed without merging**